### PR TITLE
Rework editing of templates

### PR DESCRIPTION
**Pull request by radiansz**

### CODE REVIEW

File: src/blocks/button/button.css [hunk 1]
```diff
@@ -49,11 +54,19 @@
 }
 .button_color_darkgray {
   color: #fff;
+<<<<<<< HEAD
```
Comment (radiansz) [inline src/blocks/button/button.css]: Merge traces

File: src/blocks/dropdown/letterActions/letterActions.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            'click .navigation__delete': 'delete'
+        },
+
+        initialize: function(){
+            var _super = Dropdown.prototype.initialize.apply(this, arguments);
+
+            this.context = this;
+
+            return _super;
+        },
```
Comment (radiansz) [inline src/blocks/dropdown/letterActions/letterActions.js]: New line

File: src/blocks/form/changeTemplateName/changeTemplateName.ejs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<form class="form form_createGroup">
+
+	<div class="form__body">
+		<div class="form__field">
+			<%= this.include(UI.inputText, {
```
Comment (radiansz) [inline src/blocks/form/changeTemplateName/changeTemplateName.ejs]: UI traces in this file

File: src/blocks/shared/TemplatesGallery/TemplatesGroup/TemplatesList/TemplatePreview.less [hunk 4]
```diff
@@ -0,0 +1,79 @@
+@import 'src/kit/pallete';
+
+.templatePreview {
+    backgroun-color: white;
```
Comment (radiansz) [inline src/blocks/shared/TemplatesGallery/TemplatesGroup/TemplatesList/TemplatePreview.less]: Misprint

File: src/blocks/form/changeTemplateName/changeTemplateName.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+define(function(require) {
+
+    var Form = require('blocks/form/form');
+    var app = require('app');
+    var Notification = require('blocks/notification/notification');
+
+    return Form.extend({
+        template: require('tpl!./changeTemplateName.ejs'),
```
Comment (astupak) [inline src/blocks/form/changeTemplateName/changeTemplateName.js]: Inline require here and below

File: src/pages/automation/letters/create/title.ejs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <div class="header__backtitle"><a class="navigation__link" href="/automation/letters"><%= UI.icons.back%><%=i18n('Назад к списку писем')%></a></div>
+    </div>
+    <div class="header__content">
+        <% if (letter.get('name')) { %>
+            <h1><%- letter.get('name') %></h1>
+        <% } else { %>
+            <h1><%- i18n('Создание письма') %></h1>
+        <% } %>
+        <div class="header__actions">
+            <%= this.include(require('blocks/dropdown/letterActions/letterActions'), {
```
Comment (astupak) [inline src/pages/automation/letters/create/title.ejs]: Inline require

File: src/pages/mailings/templates/edit/content.ejs [hunk 7]
```diff
@@ -1,34 +1,122 @@
...
+    </div>
+    <%= self.include(Panel, {
+        childs: [{
+            constructor: PanelItem,
+            childs: [{
+                constructor: Button,
+                childs: [{
+                    constructor: Icon,
+                    props: {
+                        icon: require('svg!kit/icons/back.svg')
```
Comment (radiansz) [inline src/pages/mailings/templates/edit/content.ejs]: Move to top

File: src/blocks/dropdown/templateActions/templateActions.js [hunk 8]
```diff
@@ -16,40 +24,85 @@ define(function(require) {
...
+                content: ChangeTemplateNameForm,
+                formModel: this.model,
+                title: i18n('Изменение названия')
+            })).show();
         },
-        delete: function(event) {
-            var block = this;
-            var confirm = new UI.confirm({
+
+        attemtToDelete: function() {
```
Comment (rosendi) [inline src/blocks/dropdown/templateActions/templateActions.js]: var

File: src/blocks/form/changeTemplateName/changeTemplateName.ejs [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+				label: i18n('Название'),
+				name: 'name',
+				autofocus: true,
+                value: this.model.get('name')
+			}) %>
+		</div>
+	</div>
+
+	<div class="form__footer">
+        <div class="form__errorMessage" for="global" style="text-align: right"></div>
```
Comment (astupak) [inline src/blocks/form/changeTemplateName/changeTemplateName.ejs]: Inline style?

File: src/blocks/form/changeTemplateName/changeTemplateName.ejs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        
+        <div class="modal__cancel">
+            <%= this.include(UI.button, {
```
Comment (astupak) [inline src/blocks/form/changeTemplateName/changeTemplateName.ejs]: UI here and below

File: src/blocks/shared/TemplatesGallery/index.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        },
+
+        constructTabs: function() {
+            var self = this;
+            var groups = this._groupTemplates();
+
+            return Object.keys(groups).map(function(subtype) {
+                var templates = groups[subtype];
+
+                return { id: subtype, label: self._getTemplatesLabel(subtype), content: TemplatesGroup, options: { templates: templates, onSetTemplate: self.props.onSetTemplate } };
```
Comment (rosendi) [inline src/blocks/shared/TemplatesGallery/index.js]: Split into separate lines

File: src/blocks/form/changeTemplateName/changeTemplateName.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+        validators: {
+            name: require('kit/validators/required')
```
Comment (radiansz) [inline src/blocks/form/changeTemplateName/changeTemplateName.js]: Requires

File: src/blocks/shared/TemplatesGallery/TemplatesGroup/TemplatesList/index.ejs [hunk 13]
```diff
@@ -0,0 +1,23 @@
+<%
+    var Cards = require('blocks/ui/cards/cards');
+    var NoTemplates = require('../NoTemplates');
+    var self = this;
+%>
+
+<div class="<%- self.generateClassName('templatesGroup__list', self.modifiers) %>">
+    <% if (self.hasTemplates() > 0) { %>
```
Comment (astupak) [inline src/blocks/shared/TemplatesGallery/TemplatesGroup/TemplatesList/index.ejs]: Why compare it with 0?

**Comment by radiansz** [comment]
LGTM :rocket:

**Comment by rosendi** [comment]
LGTM 🎶

**Comment by zouxuoz** [comment]
lgtm 👳

**Comment by astupak** [comment]
LGTM

**Pull request merged**